app/controllers/session_controller.rb

```diff
@@ -46,6 +47,10 @@ def sso_provider(payload=nil)
         else
           redirect_to sso.to_url(sso.return_sso_url)
         end
+      elsif (sso.no_user_found_return_sso_url != "" && !sso.no_user_found_return_sso_url.nil? )
```

Comment:
this block of logic shall be in `else` block, and you would want to set sso.user_found to all of them even it redirects to Discourse login page. (hence a client side plugin have the knowledge to do stuffs)